src/UCTSearch.h

```diff
@@ -89,8 +89,8 @@ class UCTSearch {
     void dump_analysis(int playouts);
     int get_best_move(passflag_t passflag);
 
-    GameState & m_rootstate;
-    UCTNode m_root{FastBoard::PASS, 0.0f, 0.5f};
+    static std::unique_ptr<GameState> m_rootstate;
```

Comment:
GameState is copied in by UCTSearch, based on the state of the UCTSearch's controller. So this should be a simple member. From looking through the code I see no clear reason why you need this to change (neither to a pointer nor to a global).

Reply:
It's needed for the conservative checks I do in find_and_take_ownership were I pass GTP's gamestate (g_new) and the Search's saved games state (g_curr).